src/main/java/org/m14n/moonthespire/cards/EbbAndFlow.java

```diff
@@ -0,0 +1,105 @@
+import com.megacrit.cardcrawl.actions.utility.HandCheckAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.cards.red.Whirlwind;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.CardStrings;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.ui.panels.EnergyPanel;
+import org.apache.logging.log4j.core.lookup.AbstractConfigurationAwareLookup;
+import org.m14n.moonthespire.MoonCharacter;
+import org.m14n.moonthespire.PhaseChangeSubscriber;
+import org.m14n.moonthespire.PhaseSystem;
+
+import static org.m14n.moonthespire.Mod.cardPath;
+import static org.m14n.moonthespire.Mod.makeId;
+
+public class EbbAndFlow extends AbstractMoonCard implements PhaseChangeSubscriber {
+    public static final String ID = makeId("EbbAndFlow");
+    public static final String IMG = cardPath("moondrop.png");
+
+    private static final int COST = 2;
+    private static final CardType TYPE = CardType.ATTACK;
+    private static final CardColor COLOR = MoonCharacter.Enums.MOON_PURPLE;
+    private static final CardRarity RARITY = CardRarity.UNCOMMON;
+    private static final CardTarget TARGET = CardTarget.ENEMY;
+    private static final CardStrings cardStrings;
+
+    public EbbAndFlow() {
+        super(ID, IMG, COST, TYPE, COLOR, RARITY, TARGET);
+        PhaseSystem.subscribe(this);
+        this.baseDamage = 5;
+        this.isMultiDamage = true;
+        this.exhaust = true;
+
+    }
+
+    @Override
+    public void onRemoveFromMasterDeck() {
+        PhaseSystem.unsubscribe(this);
+    }
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        // Create an int which equals to your current energy.
+        int count = AbstractDungeon.player.hand.size();
+
+        //Waxing
+        if (PhaseSystem.isWaxing()) {
+            // For each card, create 1 damage action.
+            for (int i = 0; i < count; i++) {
+                AbstractDungeon.actionManager.addToBottom(
+                        new DamageAction(m, new DamageInfo(p, damage, damageTypeForTurn),
+                                AbstractGameAction.AttackEffect.SLASH_HORIZONTAL));
+            }
+        }
+        //Waning
+        if (PhaseSystem.isWaning()) {
+            this.addToBot(new DrawCardAction(AbstractDungeon.player, count, false));
+        }
+
+
+        this.addToTop(new DiscardAction(AbstractDungeon.player, AbstractDungeon.player, count, false));
+    }
+
+    @Override
+    public void upgrade() {
+        if (!this.upgraded) {
+            this.upgradeName();
+            this.exhaust = false;
+            this.upgradeDamage(2);
+            this.rawDescription = cardStrings.UPGRADE_DESCRIPTION;
```

Comment:
Unnecessary

Reply:
To clarify, setting raw description is unnecessary. InitializeDescription does the work for us already